main.nf

```diff
@@ -934,8 +934,108 @@ workflow.onComplete {
+            "plotOrthologDistrib": "plot_ortholog_distributions.log", 
+            "doRateAdjustment": "rate_adjustment.log",
+            "paralogsAnalyses": "paralogs_analyses.log",
+            "drawTree": "rate_adjustment.log" ]
+
+            log_filename = "rate_adjustment/${species_name}/logs_${workflow.sessionId.toString().substring(0,8)}/${logs_names[process]}"
+
+            log_file = file("${log_filename}")
+            if ( log_file.exists() == true ) {
+                // If the process log file exists and there are "ERROR" lines
+                // (the error was caught by the script), print such lines
+                if ( log_file.readLines().findAll{ it =~ /ERROR/ }.size() != 0 ) {
+
+                    headline_error_box = "The pipeline stopped at process '${process}' with the following error message:"
+                    // Separator to highlight the beginning of the error box
+                    log.error "\n"
+                    log.error "${'=' * headline_error_box.length()}"
+                    // Write error box headline
+                    log.error "${headline_error_box}"
+                    log.error "\n"
+
+                    for( line : log_file.readLines().findAll { it =~ /ERROR/ } ) {
+                            log.error line
+                    }
+                    log.error "\n"
+                }
+                // If the process log file exists but there are no "ERROR" lines
+                // (probably it's an unexpected error), don't print any error line
+                else {
+                    headline_error_box = "The pipeline encountered an error during process '${process}'."
+                    // Separator to highlight the beginning of the error box
+                    log.error "\n"
+                    log.error "${'=' * headline_error_box.length()}"
+                    // Write error box headline
+                    log.error "${headline_error_box}"
+                    log.error "\n"
+                }
+
+                // In any case, point to the complete output of the stopped process log file
+                log.error "More details can be found in the following log file:"
+                log.error "${log_filename}"
+            }
+
+            // Else if the process log file doesn't exist, look for an external cause
+            // in errorReport and errorMessage
+            else {
+                headline_error_box = "The pipeline stopped at process '${process}' with the following error message:"
+                // Separator to highlight the beginning of the error box
+                log.error "\n"
+                log.error "${'=' * headline_error_box.length()}"
+                // Write error box headline
+                log.error "${headline_error_box}"
+                log.error "\n"
+
+                // Write the "Caused by:" line from errorReport
+                error_cause = "${workflow.errorReport.split("\n")[3]}"
```

Comment:
Again, parsing depends on specific location in `errorReport`.

Reply:
Now the code looks for a match with "Caused by:" instead of blinding taking the line by index; if "Caused by:" is not found nothing is printed.